### PR TITLE
Fix root padding for alignwide blocks.

### PR DESCRIPTION
**Pull request by tellthemachines**
Trac ticket: https://core.trac.wordpress.org/ticket/61587

Syncs changes from https://github.com/WordPress/gutenberg/pull/63207.

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/class-wp-theme-json.php [hunk 1]
```diff
@@ -2963,9 +2963,9 @@ public function get_root_layout_rules( $selector, $block_metadata ) {
 			// Alignfull children of the container with left and right padding have negative margins so they can still be full width.
 			$css .= '.has-global-padding > .alignfull { margin-right: calc(var(--wp--style--root--padding-right) * -1); margin-left: calc(var(--wp--style--root--padding-left) * -1); }';
 			// Nested children of the container with left and right padding that are not wide or full aligned do not get padding, unless they are direct children of an alignfull flow container.
```
Comment (mukeshpanchal27) [inline src/wp-includes/class-wp-theme-json.php]:
```suggestion
			// Nested children of the container with left and right padding that are not full aligned do not get padding, unless they are direct children of an alignfull flow container.
```

Comment (tellthemachines) [reply src/wp-includes/class-wp-theme-json.php]: Oh well spotted! Should update the comment on the Gutenberg side too.

Comment (mukeshpanchal27) [reply src/wp-includes/class-wp-theme-json.php]: The changes already their in GB side.

**Review by aaronrobertshaw — APPROVED** [review]
Thanks for the fix and backport @tellthemachines and @richtabor 👍 

✅ Tests as advertised (on frontend, editor will need JS package updates from GB PR)
✅ Code changes match GB PR
✅ Unit tests pass locally also

| 6.5 | 6.6 Before | 6.6 After |
|---|---|---|
| Screenshot 2024-07-08 at 3 41 41 PM | Screenshot 2024-07-08 at 3 40 58 PM | Screenshot 2024-07-08 at 3 40 18 PM |

**Comment by tellthemachines** [comment]
Committed in [r58685](https://core.trac.wordpress.org/changeset/58685).

**Review by andrewserong — COMMENTED** [review]
This change looks good to me, too, and tests nicely in the site frontend! ✅

**Pull request closed without merging**